### PR TITLE
Add back `pageContext` function

### LINKED ISSUE #26: [Discussion] universal-middleware/hono
I still don't get about universal middleware. 

I have code like this using hono

```ts
export const preference = (): MiddlewareHandler => {
  return async function preference(c, next) {
    const cookiePreference = getCookie(c, "preference")
    const preference = cookiePreference
      ? JSON.parse(cookiePreference)
      : PREFERENCE
    c.set("preference", preference)
    await next()
  }
}

export const ssr = () =>
  vikeNode({
    pageContext: (c: Context) => {
      return {
        preference: c.var.preference,
        user: c.var.user,
        vendor: c.var.vendor,
      }
    },
    static: {
      root: distClient,
    },
  })

instance.use(preference())
instance.use(ssr())
```

And with universal-middleware, can I still use hono context using native behavior like ```c.var``` and ```c.set``` ?

### PR DESCRIPTION
**Pull request by magne4000**
Fixes #26

### Goal

Do not force users to write `universal-middleware`s to update the `pageContext`

### CODE REVIEW

**Comment by brillout** [comment]
Niceee.

Let's add a `BREAKING CHANGE:`.

I'd suggest we add a `> [!WARNING] **BETA** This package will have relatively frequent breaking changes.`.

File: packages/vike-node/README.md [hunk 1]
```diff
@@ -243,17 +243,28 @@ function startServer() {
...
+You can define custom [pageContext](https://vike.dev/pageContext) properties:
 
-`vike-node` uses [universal-middleware](https://universal-middleware.dev/) and automatically adds the universal context to [`pageContext`](https://vike.dev/pageContext).
+> [!NOTE]
+> `vike-node` uses [universal-middleware](https://universal-middleware.dev/)
+> and automatically adds the universal context to [`pageContext`](https://vike.dev/pageContext).
 
-If you need custom properties to be available in `pageContext`,
-you can [create a universal context middleware](https://universal-middleware.dev/recipes/context-middleware#updating-the-context) and attach it to your server.
+```js
```
Comment (brillout) [inline packages/vike-node/README.md:254]:
```suggestion
```ts
```

**Comment by brillout** [comment]
I'd suggest we prefix all branches with `{magne4000,brillout,nitedani}`. Makes cleaning up stale branches a lot easier.

**Comment by magne4000** [comment]
> Niceee.
> 
> Let's add a `BREAKING CHANGE:`.

Why though? It does not break the current `0.2.x`

> I'd suggest we add a `> [!WARNING] **BETA** This package will have relatively frequent breaking changes.`.

:+1:

**Comment by brillout** [comment]
Ah, you're right. I guess the trick is to just update the breaking change of the `0.2.0` release and mention `Make sure to update to >= 0.2.4 `.

**Comment by brillout** [comment]
LGTM. I did a [bit of polishing](https://github.com/vikejs/vike-node/commit/485759dfbfb13a832f89e2741a14976ae1452076) (feel free to disagree).

I wonder whether the Node.js object `req: IncomingMessage` is something Hono users will expect. Maybe we should also provide the whole universal-middleware `context` & `runtime` :thinking:

**Comment by magne4000** [comment]
> Maybe we should also provide the whole universal-middleware context & runtime

It receives `runtime[runtime.adapter]` as its parameter, so `req: IncomingMessage` is just an example. It's not clear though, and the typing must currently be given by the user.
I'll update it so that the typing comes from `universal-middleware` to make things easier and less confusing.

**Comment by magne4000** [comment]
> > Maybe we should also provide the whole universal-middleware context & runtime
> 
> It receives `runtime[runtime.adapter]` as its parameter, so `req: IncomingMessage` is just an example. It's not clear though, and the typing must currently be given by the user. I'll update it so that the typing comes from `universal-middleware` to make things easier and less confusing.

Fixed by https://github.com/vikejs/vike-node/pull/29 and published as `0.2.5`

**Pull request merged**